apollo-router/src/plugins/forbid_mutations.rs

```diff
@@ -19,14 +22,21 @@ struct ForbidMutations {
     forbid: bool,
 }
 
+/// Forbid mutations configuration
+#[derive(Debug, Clone, Deserialize, Serialize, JsonSchema)]
+#[serde(deny_unknown_fields, untagged)]
+enum Bool {
```

Comment:
that feels like a lot of boilerplate for a single boolean, and I'm worried we might accidentally write the configuration for future plugins in a format that won't fit reporting.

Also, would this work with a wrapper struct like `struct ForbidMutationsConfig(bool)` ?
Because the enum result in a `anyOf` in the schema, that might be confusing https://github.com/apollographql/router/pull/2173/files#diff-ea34c7fa4e3f6984aae7f7e0d20d22930d5386763c6be9d9fb2f017adf16f746R115-R120

Reply:
I didn't want to change the config format in this PR. The issue is that bool as a config struct cannot be documented, so we have to introduce this extra layer. From the user point of view the config format doesn't change.

In future we should resist having any config that uses raw rust types directly as they can't be extended as it will require hacks like this to work around.

Introducing the enum at least keeps this self contained rather than trying to manually alter the schema elsewhere.

Reply:
right, I understand that it needs something else than primitive types for documentation, but the enum is generating a weird schema. Could you try the wrapper type `struct ForbidMutationsConfig(bool)` instead?

Reply:
[664c5ea](https://github.com/apollographql/router/pull/2173/commits/664c5ea2a03f0777c2ab6e17cc7a36e8a2eea597)